pkg/framework/kind/kind.go

```diff
@@ -0,0 +1,49 @@
+package kind
+
+import "github.com/grafana/thema"
+
+// TODO docs
+type Maturity string
+
+const (
+	MaturityCommitted    Maturity = "committed"
+	MaturitySynchronized Maturity = "synchronized"
+	MaturityMature       Maturity = "mature"
+)
+
+// TODO docs
+type Interface interface {
+	// TODO docs
+	Name() string
+
+	// TODO docs
+	Maturity() Maturity // TODO unclear if we want maturity for raw kinds
```

Comment:
Is this where you imagine having access to `models.ObjectSummaryBuilder`?

https://github.com/grafana/grafana/blob/main/pkg/services/store/kind/registry.go#L21

Reply:
Probably yes - in general, capabilities that are declared on the CUE kind framework side should have corresponding things in these interfaces (when they require a computation/behavior) or on the various `Meta` structs (when they're really just a simple config value)

The only reason it's "probably" is because, as i noted [here](https://github.com/grafana/grafana/pull/56492/files/f40576ff785baef118e222e7ff918d925bd9e42e#r992287075), we have to keep the basic generated kinds separated from the bulk of Grafana dependencies. For kinds with a go passthrough func to call for the object summary builder, i suspect that will have dependencies that violate that rule. 

The solution is basically just more layers/veneering - it's perfectly fine for us to generate the dependency-needing code under `pkg/service` in some disciplined way. Maybe that'll also mean a separate interface, too, that layers on top of these. It may even make the most sense to just handle all the object summary production outside of `pkg/kind`, as that's not something which we expect to be useful in any as-code cases, anyway.